### PR TITLE
Realtime segment size threshold metrics

### PR DESCRIPTION
**Pull request by mcvsubbu**
Added a couple of metrics for observing the actual segment size and the predicted number of rows we indicate to the servers. The metrics should help us improve the prediction algorithm for new use cases.

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentSizeBasedFlushThresholdUpdater.java [hunk 1]
```diff
@@ -50,5 +57,9 @@ public synchronized void updateFlushThreshold(StreamConfig streamConfig, Segment
         _flushThresholdComputer.computeThreshold(streamConfig, committingSegmentDescriptor, committingSegmentZKMetadata,
             newSegmentZKMetadata.getSegmentName());
     newSegmentZKMetadata.setSizeThresholdToFlushSegment(threshold);
+
+    _controllerMetrics.setOrUpdateTableGauge(_rawTableName, ControllerGauge.NUM_ROWS_THRESHOLD, threshold);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentSizeBasedFlushThresholdUpdater.java:61]: Do we want to log it under the raw name or real-time table name (with suffix)? Seems most of the gauges are under the table name with suffix

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentSizeBasedFlushThresholdUpdater.java (hunk 1 above)
Comment (mcvsubbu) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentSizeBasedFlushThresholdUpdater.java:61]: I thought about it and decided on raw name. But I don't have a preference one way or another.

**Review by sajjad-moradi — APPROVED** [review]
LGTM

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java [hunk 2]
```diff
@@ -165,6 +165,16 @@ public enum ControllerGauge implements AbstractMetrics.Gauge {
...
 
+  // A per-table metric that shows the number of rows we expect to consume for the next segment of
+  // any partition in the realtime table. This metric is emitted from the segment size based threshold
+  // computer.
+  NUM_ROWS_THRESHOLD("numRowsThreshold", false),
+
+  // The actual segment size for committing segments. These may be shorter than expected when the administrator
+  // issues a force-commit, or zero when new partitions are detected in the stream (since there is no completing
+  // segment when the partition is first detected).
+  COMMITTING_SEGMENT_SIZE("numRowsThreshold", false),
```
Comment (sajjad-moradi) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java:176]: Although `unit`, the first param, is not used anywhere in the code base, let's not use "numRowsThreshold" to be consistent with other metrics.

Comment (sajjad-moradi) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java:176]: Although `unit`, the first param, is not used anywhere in the code base, let's not use "numRowsThreshold" to be consistent with other metrics.

Comment (mcvsubbu) [reply pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java:176]: Good catch, it was a cut-paste problem

**Pull request merged**